sbi/inference/base.py

```diff
@@ -152,6 +153,42 @@ def __init__(
             median_observation_distances=[], epochs=[], best_validation_log_probs=[],
         )
 
+    def _append_training_data(
+        self,
+        theta: Tensor,
+        x: Tensor,
+        external_data: Tuple[Tensor, Tensor],
+        round: int,
+        exclude_invalid_x: bool,
+    ) -> None:
+        """
+        Take simulation data and external data, and merge them into attributes.
+
+        Args:
+            theta: Simulation parameters.
+            x: Simulation outputs.
+            external_data: Externally provided data consisting of (theta,x).
+            round: Current round.
+            exclude_invalid_x: Whether to exclude simulation outputs `x=NaN` or `x=±∞`
+                during training. Expect errors, silent or explicit, when `False`.
+        """
+
+        # In the first round, the user can externally provide data.
+        if external_data is not None and round == 0:
+            theta = torch.cat((external_data[0], theta))
+            x = torch.cat((external_data[1], x))
+
+        # Check for NaNs in simulations.
+        is_valid_x, num_nans, num_infs = handle_invalid_x(x, exclude_invalid_x)
+        warn_on_invalid_x(num_nans, num_infs, exclude_invalid_x)
+
+        # XXX Rename bank -> rounds/roundwise.
```

Comment:
Could be a good point in time to address this

Reply:
any suggestions here? `theta_rounds` would be an option, but not sure if I like it more than `theta_bank`.

Reply:
How about `_theta_round_data` or `_theta_round_bank` (same for `x`)?

Reply:
`_roundwise_thetas`. I think `bank` is not very clear and I find `data` too uninformative -- everything is `data`.